### PR TITLE
Update TrezorConnect API to v7

### PR DESCRIPTION
**Pull request by sarveshwar-singh**
REF: https://github.com/Emurgo/yoroi-frontend/issues/277

Wanted to remove `trezorProtocolMagic` but there is no staging Protocol Magic for Trezor so left it like that. (in review its decided that we should remove `trezorProtocolMagic` and use old `protocolMagic` config property)

### CODE REVIEW

File: config/config-types.js [hunk 1]
```diff
@@ -24,9 +24,9 @@ export type NetworkConfigType = {
   backendUrl: string,
   websocketUrl: string,
   name: Network,
-  trezorNetwork:
-  1 // testnet
-  | 2, // Mainnet
+  trezorProtocolMagic:
+    764824073   // mainnet
+  | 1097911063, // testnet
```
Comment (nicarq) [inline config/config-types.js]: I would change the wording from `testnet` to `staging` (there is a testnet chain and it uses a diff protocolMagic)

File: docs/specs/code/TREZOR_INTEGRATION.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+# Iteration-1
+
+# Proposal
+User will be able to:
+1. Setup a new Yoroi Wallet without exposing its private key/ mnemonics.
+2. Send ADA using the Trezor Wallet Security.
+
+## Prerequisite
+
+1. Trezor Model T - Version 2.0.8 or later. (Support for Trezor One is supposed to come by mid-next year according to some information in Reddit).
```
Comment (nicarq) [inline docs/specs/code/TREZOR_INTEGRATION.md]: I think it should be 2.1.0

File: docs/specs/code/TREZOR_INTEGRATION.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
+# Abstract
+
+1. Users would be able to use a Trezor Hardware Wallet with Yoroi Wallet.
+2. Hardware wallet can be integrated by creating new Yoroi wallet, by using `Connect to Trezor Hardware Wallet` on `Add Wallet` page.
+3. [TrezorConnect](https://github.com/trezor/connect) API will be used to communicate with hardware device.
+
+# Motivation
+
+1. Since private key never leave the hardware wallet so it's considered as one of the most secured way of using cryptocurrency wallets.
+2. No need to remember spending passowrd, so its easy to use.
```
Comment (nicarq) [inline docs/specs/code/TREZOR_INTEGRATION.md]: I would mention that you still need to remember a pincode, so it’s easier but not necessarily easy.

Comment (sarveshwar-singh) [reply docs/specs/code/TREZOR_INTEGRATION.md]: @nicarq 
we can add that in drawback block.

File: docs/specs/code/TREZOR_INTEGRATION.md (hunk 2 above)
Comment (nicarq) [reply docs/specs/code/TREZOR_INTEGRATION.md]: deleting this one, because it’s off.

**Comment by sarveshwar-singh** [comment]
@nicarq 
Review fixes are done.

cc @SebastienGllmt

File: app/config.js [hunk 4]
```diff
@@ -17,6 +17,10 @@ export default {
   trezor: {
     // https://github.com/trezor/connect/blob/develop/docs/methods/cardanoGetPublicKey.md
     // https://cardanolaunch.com/assets/Ed25519_BIP.pdf
-    DEFAULT_CARDANO_PATH: 'm/44\'/1815\'/0\''
+    DEFAULT_CARDANO_PATH: 'm/44\'/1815\'/0\'',
+    manifest: {
+      EMAIL: 'systems@emurgo.io',
+      APP_URL: 'https://chrome.google.com/webstore/detail/yoroi/ffnbelfdoeiohenkjibnmadjiehjhajb'
```
Comment (SebastienGllmt) [inline app/config.js]: this will not be accurate for Firefox (although I don't think Trezor works on Firefox anyway?)

File: config/testnet.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
     "backendUrl": "https://18.207.22.26:8080",
     "websocketUrl": "wss://18.207.22.26:8080",
     "name": "testnet",
-    "trezorNetwork": 1
+    "trezorProtocolMagic": 1097911063
```
Comment (SebastienGllmt) [inline config/testnet.json]: testnet is not the same as staging so it should have a different magic

File: app/config.js (hunk 4 above)
Comment (sarveshwar-singh) [reply app/config.js]: https://github.com/Emurgo/yoroi-frontend/issues/344

File: config/config-types.js (hunk 1 above)
Comment (nicarq) [reply config/config-types.js]: we should also change the protocolMagic to `633343913`

File: config/staging.json [hunk 6]
```diff
@@ -4,7 +4,7 @@
     "backendUrl": "https://iohk-staging.yoroiwallet.com",
     "websocketUrl": "wss://iohk-staging.yoroiwallet.com:443",
     "name": "staging",
-    "trezorNetwork": 1
+    "trezorProtocolMagic": 1097911063
```
Comment (nicarq) [inline config/staging.json]: this one should be `633343913`

**Pull request merged**